src/components/map-control.tsx

```diff
@@ -58,7 +58,7 @@ export const MapControl = ({children, position}: MapControlProps) => {
     controls.push(controlContainer);
 
     return () => {
-      const index = controls.getArray().indexOf(controlContainer);
+      const index = controls.getArray()?.indexOf(controlContainer);
```

Comment:
I think I like this a bit more explicit and with a short explanation so that no one (me) will "optimize" that away again :)

```suggestion
      const controlsArray = controls.getArray();
      // controlsArray could be undefined if the map is in an undefined state (e.g. invalid API-key, see #276
      if (!controlsArray) return;

      const index = controlsArray.indexOf(controlContainer);
```